src/background_music/FrameBasedEffectPlugin.h

```diff
@@ -0,0 +1,190 @@
+#ifndef AUDACIOUS_PLUGINS_FRAMEBASEDEFFECTPLUGIN_H
+#define AUDACIOUS_PLUGINS_FRAMEBASEDEFFECTPLUGIN_H
+/*
+ * audacious-plugins/FrameBasedEffectPlugin.h
+ *
+ * Added by michel on 2023-09-23
+ * Copyright (C) 2015-2023 Michel Fleur.
+ * Source https://github.com/emmef/audacious-plugins
+ * Email audacious-plugins@emmef.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <libaudcore/plugin.h>
+#include <libaudcore/ringbuf.h>
+
+#include "Detection.h"
+
+template<class D>
+class FrameBasedEffectPlugin : public EffectPlugin
```

Comment:
This does not need to be a template class. Only one class is ever used to satisfy the template parameter D.